src/plugins/intel_gpu/src/graph/mutable_data.cpp

```diff
@@ -81,4 +81,42 @@ mutable_data_inst::typed_primitive_inst(network& network, mutable_data_node cons
     }
 }
 
+void mutable_data_inst::save(cldnn::BinaryOutputBuffer& ob) const {
+    parent::save(ob);
+
+    const auto _allocation_type = _outputs[0]->get_allocation_type();
+    ob << make_data(&_allocation_type, sizeof(_allocation_type));
+
+    size_t data_size = _outputs[0]->size();
+    ob << make_data(&data_size, sizeof(size_t));
+
+    if (_allocation_type == allocation_type::usm_host || _allocation_type == allocation_type::usm_shared) {
+        ob << make_data(_outputs[0]->buffer_ptr(), data_size);
+    } else {
+        mem_lock<char, mem_lock_type::read> lock{_outputs[0], get_node().get_program().get_stream()};
+        ob << make_data(lock.data(), data_size);
+    }
+}
+
+void mutable_data_inst::load(BinaryInputBuffer& ib) {
+    parent::load(ib);
+
+    allocation_type _allocation_type;
+    ib >> make_data(&_allocation_type, sizeof(_allocation_type));
+
+    size_t data_size;
+    ib >> make_data(&data_size, sizeof(size_t));
+
+    OPENVINO_ASSERT(_outputs[0] != nullptr, "Output memory should be allocated before importing data.");
+
+    if (_allocation_type == allocation_type::usm_host || _allocation_type == allocation_type::usm_shared) {
```

Comment:
Do we really need to serialize the allocation type? Can't we use just type of allocated _outputs[0] buffer? Also, it will exclude possible situation with mismatch of allocation type between allocated buffer and serialized.

Reply:
I updated to use the type of `_outputs[0]` buffer, instead of serializing it. Thank you!